### PR TITLE
Gas customization features

### PR DESCRIPTION
**Pull request by danjm**
Ready for final QA

### CODE REVIEW

File: ui/app/selectors/custom-gas.js [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    addHexPrefix
+  )(price)
+}
+
+function getRenderableBasicEstimateData (state) {
+  if (getBasicGasEstimateLoadingStatus(state)) {
+    return []
+  }
+  const gasLimit = state.metamask.send.gasLimit || getCustomGasLimit(state)
+  console.log('--------------gasLimit--------------', gasLimit)
```
Comment (tmashuang) [inline ui/app/selectors/custom-gas.js]: `Console.log` here.

**Comment by danjm** [comment]
@tmashuang Thanks. I've fixed those now.

**Comment by tmashuang** [comment]
Screen is crowded when Hex Data is enabled.
screenshot from 2018-11-30 12-57-02

**Comment by danjm** [comment]
@tmashuang That issue is fixed

peek 2018-12-03 20-22

**Review by alextsg — APPROVED** [review]

**Pull request merged**